sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -128,6 +129,15 @@ case class CreateViewCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
+    // CREATE VIEW AS INSERT INTO ... is not allowed, we should throw an AnalysisException.
+    analyzedPlan match {
+      case i: InsertIntoHadoopFsRelationCommand =>
```

Comment:
Hmmm... fixing this in grammar itself would a little bit of work (we also support multi-insert and that makes this harder). I suppose we could try to add a check in the `AstBuilder`.

Why not check this in the `CreateViewCommand`? That seems safer to me.